mcf-browser/src/back-end/data.ts

```diff
@@ -0,0 +1,250 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const ID_DELIMITER = ',';
+
+const ID_PROPERTIES = [
+  'variableMeasured',
+  'observationAbout',
+  'provenance',
+  'measurementMethod',
+  'observationPeriod',
+  'unit',
+  'scalingFactor',
+] as const;
+type IdProperties = typeof ID_PROPERTIES[number];
+type SeriesIdObject = Pick<Series, IdProperties>;
+
+type DataPoint = {
+  x: string;
+  y: number;
+};
+
+/** Class representation of a single series */
+class Series {
+  /** A unique identifier made using the facet
+  */
+  id: string;
+
+  /** An ordered list of datapoints */
+  data: DataPoint[];
+
+  /** The observationAbout of the data */
+  observationAbout?: string;
+
+  /** The variable being measured */
+  variableMeasured?: string;
+
+  /** The provenance of the data */
+  provenance?: string;
+
+  /** The measurement method of the data */
+  measurementMethod?: string;
+
+  /** The length of time over which the data point was collected */
+  observationPeriod?: string;
+
+  /** The unit for the data */
+  unit?: string;
+
+  /** The scaling factor for the data */
+  scalingFactor?: number;
+
+  /**
+   * Constructor for the series class, creates the object
+   * and instantiates all of the properties
+   * @param {DataPoint[]} data the datapoints for the series
+   * @param {string} variableMeasured the variableMeasured of a series
+   * @param {string} observationAbout the observationAbout of a series
+   * @param {string} provenance the provenance of a series
+   * @param {string} measurementMethod the measurementMethod of a series
+   * @param {string} observationPeriod the observationPeriod of a series
+   * @param {string} unit the unit of a series
+   * @param {number} scalingFactor the scalingFactor of a series
+   */
+  constructor(
+      data: DataPoint[],
+      variableMeasured?: string,
+      observationAbout?: string,
+      provenance?: string,
+      measurementMethod?: string,
+      observationPeriod?: string,
+      unit?: string,
+      scalingFactor?: number,
+  ) {
+    const cleanVariable = (variable: string | undefined) => {
+      if (!variable) {
+        return variable;
+      } else if (variable.startsWith('dcs:')) {
+        return variable.slice(4);
+      } else if (variable.startsWith('dcid:')) {
+        return variable.slice(5);
+      } else if (variable.startsWith('schema:')) {
+        return variable.slice(7);
+      }
```

Comment:
Consider using constants for `dcs:` and try to avoid hardcoding lengths... for e.g.,

```
NAMESPACE_PREFIXES = ['dcs:', 'dcid:', 'schema:'];

...

for (var prefix of NAMESPACE_PREFIXES) {
  if (variable.startsWith(prefix)) {
    return variable.slice(prefix.length);
  }
}
```

Reply:
Good point, it's changed in the next commit!